File: packages/jexlang-java/src/main/java/com/jexlang/java/types/JexBoolean.java
package com.jexlang.java.types;

public class JexBoolean implements JexValue {
    private final boolean value;
    public JexBoolean(boolean value) { this.value = value; }

    @Override
    public String getType() {
        return "boolean";
    }

    @Override
    public String toString() {
        return value ? "true" : "false";
    }

    @Override
    public Object toObject() {
        return value;
    }

    public boolean isNumber() { return false; }
    public boolean isBoolean() { return true; }
    public boolean isString() { return false; }
    public boolean isNull() { return false; }
    public boolean isArray() { return false; }
    public boolean isObject() { return false; }
    public Number asNumber(String ctx) { throw JexValue.typeError("number", ctx, this); }
    public boolean asBoolean(String ctx) { return value; }
    public String asString(String ctx) { throw JexValue.typeError("string", ctx, this); }
    public java.util.List<JexValue> asArray(String ctx) { throw JexValue.typeError("array", ctx, this); }
    public java.util.Map<String, JexValue> asObject(String ctx) { throw JexValue.typeError("object", ctx, this); }
}

File: packages/jexlang-swift/jexlang-swift/eval/types/JexBoolean.swift
//
//  JexBoolean.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 02/11/25.
//

import Foundation

public class JexBoolean: JexValue {
    public func isInteger() -> Bool {
        return false
    }
    
    public func isDouble() -> Bool {
        return false
    }
    
    public func isBoolean() -> Bool {
        return true
    }
    
    public func isNumber() -> Bool {
        return false
    }
    
    public func isString() -> Bool {
        return false
    }
    
    public func isNil() -> Bool {
        return false
    }
    
    public func isArray() -> Bool {
        return false
    }
    
    public func isObject() -> Bool {
        return false
    }
    
    public func getType() -> String {
        return "boolean"
    }
    
    public func toObject() -> AnyObject? {
        return value as AnyObject
    }
    
    public func asDouble(context: String) throws -> Double {
        throw JexValueFactory.typeError(want: "double", ctx: context, actualValue: self)    }
    
    public func asInteger(context: String) throws -> Int {
        throw JexValueFactory.typeError(want: "double", ctx: context, actualValue: self)
    }
    
    public func asNumber(context: String) throws -> NSNumber {
        throw JexValueFactory.typeError(want: "number", ctx: context, actualValue: self)
    }
    
    public func asBoolean(context: String) throws -> Bool {
        return value
    }
    
    public func asString(context: String) throws -> String {
        throw JexValueFactory.typeError(want: "string", ctx: context, actualValue: self)
    }
    
    public func asArray(context: String) throws -> [any JexValue] {
        throw JexValueFactory.typeError(want: "array", ctx: context, actualValue: self)
    }
    
    public func asObject(context: String) throws -> [String : any JexValue] {
        throw JexValueFactory.typeError(want: "object", ctx: context, actualValue: self)
    }
    
    private let value: Bool
    
    internal init(value: Bool) {
        self.value = value
    }
    
    public var description: String {
        return "\(value)"
    }
    
    public func equals(_ other: JexValue) -> Bool {
        guard other.isBoolean() else { return false }
        return (try? other.asBoolean(context: "equals")) == value
    }
}
